webhooks/virtualmachine/v1alpha2/validation/virtualmachine_validator.go

```diff
@@ -349,28 +352,48 @@ func (v validator) validateStorageClass(ctx *context.WebhookRequestContext, vm *
 	scPath := field.NewPath("spec", "storageClass")
 	scName := vm.Spec.StorageClass
 
-	// TODO: This validation shouldn't be done in the webhook.
-
-	sc := &storagev1.StorageClass{}
-	if err := v.client.Get(ctx, client.ObjectKey{Name: scName}, sc); err != nil {
-		return append(allErrs, field.Invalid(scPath, scName, fmt.Sprintf(storageClassNotFoundFmt, vm.Namespace)))
+	var zones []topologyv1.AvailabilityZone
+	if pkgconfig.FromContext(ctx).Features.PodVMOnStretchedSupervisor {
+		azs, err := topology.GetAvailabilityZones(ctx, v.client)
+		if err != nil {
+			return append(allErrs, field.Invalid(scPath, scName, fmt.Sprintf(storageClassNotFoundFmt, vm.Namespace)))
```

Comment:
Can we return diff error. `storageClassNotFoundFmt` is not the right error when failed to get AZs.